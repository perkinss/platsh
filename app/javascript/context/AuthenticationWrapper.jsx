import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
