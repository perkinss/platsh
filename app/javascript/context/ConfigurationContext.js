import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
