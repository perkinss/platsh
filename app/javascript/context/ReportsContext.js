import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
