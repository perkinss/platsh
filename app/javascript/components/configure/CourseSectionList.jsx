import React, { useContext, useEffect } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Table, TableBody, TableCell, TableHead, TableRow, Paper, Typography, Button } from '@material-ui/core';
import MaterialTable from 'material-table'
import moment from 'moment'
import { ConfigurationContext } from "../../context/ConfigurationContext";
import timeSince from "../../helpers/date_helper";

const styles = theme => ({
    root: {
        width: '100%',
        marginTop: theme.spacing(3),
        overflowX: 'auto',
    },
    table: {
        minWidth: 700,
    },
});

function CourseSectionList(props) {
    const { classes } = props;
    const { state, dispatch } = useContext(ConfigurationContext)
    const sections = state.sections

    const formatCourses = (row) => {
         return row.courses.reduce((courses, course) => {
               return courses === '' ? course.title : courses + ", " + course.title
            }, '')
    }

    const formatGrades = (row) => {
        return row.courses.reduce((grades, grade) => {
            return grades === '' ? grade.grade : grades + ", " + grade.grade
        }, '')
    }

    const handleSelectRow = (event, section) => {
        dispatch({type:'SET_SELECTED_SECTION', section: section})
    }

    const handleDeleteRow = (event, section) => {
        let token = document.querySelector('head>meta[name="csrf-token"]').content

        fetch(`api/sections/${section.id}`,
            {
                method: 'DELETE',
                headers: {
                    'X-Requested-With': 'XMLHttpRequest',
                    'X-CSRF-Token': token,
                    'Content-Type': 'application/json',
                    'Accept': 'application/json'
                },
                credentials: 'same-origin'
            })
            .then(res => res.json())
            .then(response => {
                dispatch({type:'REMOVE_SECTION', id: section.id})
            })
            .catch(error => console.log("ERROR: ", error))
    }

    const filterCourseTitles = (term, rowData) => {
        return rowData.courses.find( course => course.title.search(term) >= 0 )
    }

    const filterGrades = (term, rowData) => {
        return rowData.courses.find ( course => course.grade && course.grade.search(term) >=0 )
    }

    const filterEnrollment = (term, rowData) => {
        if (Number(term)) {
            return rowData.students.length === Number(term)
        } else {
            let foundStudents = state.students.filter( student => Object.keys(rowData.students).includes(`${student.id}`))
            return foundStudents.find ( student => student.name.toLowerCase().search(term.toLowerCase()) >=0 )
        }
    }

    return (
        <React.Fragment>
            {sections && sections.length > 0 &&
            <Paper className={classes.root}>
                <MaterialTable
                    options={{filtering: true}}
                    actions={[
                        {
                            icon: 'edit',
                            tooltip: 'Edit Section',
                            onClick: (event, rowData) => {
                                handleSelectRow(event, rowData)
                            }
                        },
                        rowData => ({
                            icon: 'delete',
                            tooltip: rowData.assessments ? rowData.assessments.length > 0 ? '' : 'Delete Section' : '',
                            disabled: rowData.assessments ? rowData.assessments.length > 0 : false,
                            onClick: (event, rowData) => {
                                if (confirm("You sure you want to delete?")) {
                                    handleDeleteRow(event, rowData)
                                }
                            }
                        })
                    ]}
                    className={classes.table}
                    title={"Sections"}
                    data={sections}
                    columns={[
                        {title: 'Name', field: 'name'},
                        {
                            title: 'Courses',
                            field: 'courses',
                            render: rowData => formatCourses(rowData),
                            customFilterAndSearch: filterCourseTitles,
                        },
                        {
                            title: 'Grades',
                            field: 'grades',
                            render: rowData => formatGrades(rowData),
                            customFilterAndSearch: filterGrades,
                        },
                        {
                            title: 'Enrollment',
                            field: 'students',
                            render: rowData => rowData.students ? rowData.students.length : '0',
                            customFilterAndSearch: filterEnrollment,
                        },
                        {
                            title: 'Assessments',
                            field: 'assessments',
                            render: rowData => rowData.assessments ? rowData.assessments.length : '0'
                        },
                        {
                            title: 'Updated',
                            field: 'updated_at',
                            render: rowData => timeSince(new Date(rowData.updated_at)),
                            filtering: false
                        }
                    ]}/>

            </Paper>
            }
        </React.Fragment>
    );
}

CourseSectionList.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(CourseSectionList);