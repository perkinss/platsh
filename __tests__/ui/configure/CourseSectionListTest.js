import React from 'react'
import CourseSectionList from '../../../app/javascript/components/configure/CourseSectionList'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import { ConfigurationContext } from '../../../app/javascript/context/ConfigurationContext'

Enzyme.configure({ adapter: new Adapter() });

describe('<CourseSectionList />', () => {
    let render;
    let configuration = {
        sections: []
    }
    let dispatch = () =>{}

    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ConfigurationContext.Provider value={{dispatch: dispatch, state: configuration}} >
                <CourseSectionList />
            </ConfigurationContext.Provider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

  // Test with a data fetch somehow

})
