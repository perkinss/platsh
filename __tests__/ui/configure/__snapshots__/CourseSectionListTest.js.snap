// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseSectionList /> should match the snapshot 1`] = `
Object {
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
